### ISSUE TITLE
Tracking Issue for Accessibility Support

### ISSUE THREAD

**Issue by geom3trik** [vizia/vizia#56]
- [x] #55 
 - [ ] Support for screen readers

**Comment by baconpaul** [comment]
https://github.com/AccessKit/accesskit

Of interest?

**Comment by baconpaul** [comment]
ok cool. sorry thought i would share in case you didn't know; we were talking about this in context of audio plugins and thought i would share this link if you hadn't seen it.

**Comment by baconpaul** [comment]
The JUCE implementation of accessibility, although C++ based, is fantastic by the way. If you are looking for cross-platform os neutral implementations which are working.

**Comment by geom3trik** [comment]
Ah that's awesome that you've made your synthesizer accessible and have attracted this community around it. I know how important accessibility is and it's definitely one of my goals for Vizia. Unfortunately there's not a lot of Rust crates in this space at the moment, bar accesskit which is incomplete, but hopefully soon I can dedicate some time to the problem. 

> The JUCE implementation of accessibility, although C++ based, is fantastic by the way. If you are looking for cross-platform os neutral implementations which are working.

I'll definitely take a look at this for some reference but from what I understand interfacing C++ with Rust is very difficult to do at the moment.

**Comment by baconpaul** [comment]
Yes obviously juce is useful for reference only. Their model couples with their component model

but from having implemented this there’s a small set of things you can do to get to “ok” (tab order, labels, report types, good keybindings, invalidation and announcement are the key ones I think) which mean you can imagine a compact starter api helping. Always happy to chat if that’s of use

**Comment by robbert-vdh** [comment]
JUCE sadly doesn't seem to support [AT-SPI](https://en.wikipedia.org/wiki/Assistive_Technology_Service_Provider_Interface), does it? That would be a useful implementation to have a reference for in the context of (plugin) GUIs, since that's platform neutral.

**Comment by baconpaul** [comment]
i believe no, it just supports mac and win. but i could ask em if you want. when i looked accesibility on linux was not awesome

**Comment by geom3trik** [comment]
Accesskit has released a platform adapter for winit: <https://crates.io/crates/accesskit_winit>. It should now be possible to integrate accesskit without needing any changes to winit.

**Comment by RastislavKish** [comment]
Hello! 👋

Thank you for working on accessibility! I wonder, has there been any progress on this since the accesskit implementation? As far as I can tell, accesskit is now being used in the default feature-set. But if I run the example counter app, the window behaves like the a11y support has not been turned on.

Testing with Orca 49.rc on GNOME 48.4 Wayland, openSUSE Tumbleweed.
